benchmark/python/sparse/sparse_memory_benchmark.py

```diff
@@ -0,0 +1,76 @@
+"""Should be run with valgrind to get memory consumption
+   for sparse format storage and dot operators"""
```

Comment:
nit: mention that this is for CPU mem in the doc
nit: rename file to `memory_benchmark.py`?

Reply:
Renamed the file to memory_benchmark.py and  added info for CPU mem